Gordon360/ApiControllers/HousingController.cs

```diff
@@ -209,14 +218,21 @@ public IHttpActionResult EditApplication(int applicationID, [FromBody] Apartment
                 throw new BadInputException() { ExceptionMessage = errors };
             }
             //get token data from context, username is the username of current logged in person
-            ClaimsPrincipal authenticatedUser = this.ActionContext.RequestContext.Principal as ClaimsPrincipal;
+            ClaimsPrincipal authenticatedUser = ActionContext.RequestContext.Principal as ClaimsPrincipal;
             string username = authenticatedUser.Claims.FirstOrDefault(x => x.Type == "user_name").Value;
 
             string sessionID = Helpers.GetCurrentSession().SessionCode;
 
-            string newEditorUsername = applicationDetails.EditorUsername;
+            string newEditorUsername = applicationDetails?.EditorProfile?.AD_Username ?? applicationDetails?.EditorUsername;
 
             ApartmentApplicantViewModel[] newApartmentApplicants = applicationDetails.Applicants;
+            foreach (ApartmentApplicantViewModel applicant in newApartmentApplicants)
+            {
+                if (applicant.Profile == null)
+                {
+                    applicant.Profile = _profileService.GetStudentProfileByUsername(applicant?.Profile?.AD_Username ?? applicant?.Username);
```

Comment:
Same as above, `applicant.Profile` is guaranteed to be null in this context.

Reply:
```suggestion
                    applicant.Profile = _profileService.GetStudentProfileByUsername(applicant.Username);
```
That null check on `applicant` is also redundant since the code would never execute this scope without an applicant object